services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -125,6 +143,129 @@ public Optional<Long> totalSizeInBytes() {
         return Optional.ofNullable(totalSizeInBytes);
     }
 
+    /**
+     * Persists this download object to a file in Base64-encoded JSON format.
+     *
+     * @param path The path to the file to which you want to write the serialized download object.
+     * @throws IOException If the data cannot be persisted
+     */
+    public void writeToFile(Path path) throws IOException {
+        Files.write(path, toBase64Bytes(ResumableFileDownloadSerializer.toJson(this)));
+    }
+
+    /**
+     * Writes the serialized JSON data representing this object to an output stream.
+     *
+     * @param outputStream The output stream to write the serialized object to.
+     * @throws IOException If the data cannot be copied from the input to output stream
+     */
+    public void writeToOutputStream(OutputStream outputStream) throws IOException {
+        byte[] bytes = ResumableFileDownloadSerializer.toJson(this);
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes);
+        IoUtils.copy(byteArrayInputStream, outputStream);
+        IoUtils.closeQuietly(byteArrayInputStream, null);
```

Comment:
We probably should not close outputstream to be consistent with Responsetransformer#toOutputStream

https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/ResponseTransformer.java#L150

Reply:
I had missed that, will change.